lib/kernel/src/gen_tcp_socket.erl

```diff
@@ -2817,19 +2819,19 @@ handle_recv_error(P, D, ActionsR, Reason) ->
     %% ?DBG({P#params.socket, Reason}),
     {D_1, ActionsR_1} =
         cleanup_recv_reply(P, D#{buffer := <<>>}, ActionsR, Reason),
-    case Reason of
-        closed ->
-            {next_state, 'closed_read', {P, D_1}, reverse(ActionsR_1)};
-        econnreset ->
-            _ = socket_close(P#params.socket),
-            {next_state, 'closed', {P, D_1}, reverse(ActionsR_1)};
-        econnaborted ->
-            _ = socket_close(P#params.socket),
-            {next_state, 'closed', {P, D_1}, reverse(ActionsR_1)};
-        emsgsize ->
+    if
```

Comment:
There are not many cases when using `if` is relevant in Erlang code. I think this is not the suitable case and adds a bad styling into the Erlang core code!
http://rvirding.blogspot.com/2008/11/erlang-rationale-problem-with-if.html

Reply:
I think this is a suitable case since the conditions are simple comparisons against atoms (with no risk of a surprising failure that skips the clause) and the purpose is to re-use clause bodies which increases readability.